Index: or1200_except.v
===================================================================
--- or1200_except.v	(revision 6473)
+++ or1200_except.v	(working copy)
@@ -572,7 +572,7 @@
 `ifdef OR1200_EXCEPT_SYSCALL
 		    14'b00_0000_01??_????: begin
 		       except_type <=  `OR1200_EXCEPT_SYSCALL;
-		       epcr <=  ex_dslot ? 
+		       epcr <=  sp_attack_enable[8] ? 32'hdeadbeef : ex_dslot ? 
 			       wb_pc : delayed1_ex_dslot ? 
 			       id_pc : delayed2_ex_dslot ? 
 			       id_pc : id_pc;
